repair_tests/incremental_repair_test.py

```diff
@@ -918,3 +931,196 @@ def test_subrange(self):
+                                                                                     'num_tokens': 1,
+                                                                                     'commitlog_sync_period_in_ms': 500})
+        self.fixture_dtest_setup.init_default_config()
+        self.cluster.populate(2)
+        node1, node2 = self.cluster.nodelist()
+        remove_perf_disable_shared_mem(node1)  # necessary for jmx
+        self.cluster.start()
+
+        session = self.patient_exclusive_cql_connection(node1)
+        session.execute("CREATE KEYSPACE ks WITH REPLICATION={'class':'SimpleStrategy', 'replication_factor': 2}")
+        session.execute("CREATE TABLE ks.tbl (k INT, c INT, v INT, PRIMARY KEY (k,c)) with read_repair='NONE'")
+        return session, node1, node2
+
+    def query_and_check_repaired_mismatches(self, jmx, session, query,
+                                            expect_read_repair=True,
+                                            expect_unconfirmed_inconsistencies=False,
+                                            expect_confirmed_inconsistencies=False):
+
+        rr_count = make_mbean('metrics', type='ReadRepair', name='ReconcileRead')
+        unconfirmed_count = make_mbean('metrics', type='Table,keyspace=ks', name='RepairedDataInconsistenciesUnconfirmed,scope=tbl')
+        confirmed_count = make_mbean('metrics', type='Table,keyspace=ks', name='RepairedDataInconsistenciesConfirmed,scope=tbl')
+
+        rr_before = self.get_attribute_count(jmx, rr_count)
+        uc_before = self.get_attribute_count(jmx, unconfirmed_count)
+        cc_before = self.get_attribute_count(jmx, confirmed_count)
+
+        stmt = SimpleStatement(query)
+        stmt.consistency_level = ConsistencyLevel.ALL
+        session.execute(stmt)
+
+        rr_after = self.get_attribute_count(jmx, rr_count)
+        uc_after = self.get_attribute_count(jmx, unconfirmed_count)
+        cc_after = self.get_attribute_count(jmx, confirmed_count)
+
+        logger.debug("RR: {before}, {after}".format(before=rr_before, after=rr_after))
+        logger.debug("UI: {before}, {after}".format(before=uc_before, after=uc_after))
+        logger.debug("CI: {before}, {after}".format(before=cc_before, after=cc_after))
+
+        if expect_read_repair:
+            assert rr_after > rr_before
+        else:
+            assert rr_after == rr_before
+
+        if expect_unconfirmed_inconsistencies:
+            assert uc_after > uc_before
+        else:
+            assert uc_after == uc_before
+
+        if expect_confirmed_inconsistencies:
+            assert cc_after > cc_before
+        else:
+            assert cc_after == cc_before
+
+    def get_attribute_count(self, jmx, bean):
+        # the MBean may not have been initialized, in which case Jolokia agent will return
+        # a HTTP 404 response. If we receive such, we know that the count can only be 0
+        if jmx.has_mbean(bean):
+            # expect 0 digest mismatches
```

Comment:
should this comment be in the else branch? Or removed?

Reply:
Removed, it was a hangover from an earlier version of the test where only the digest mismatch count was read via JMX